### PR TITLE
update revoke to use the nuke endpoint

### PR DESCRIPTION
**Pull request by oconnor663**
@maxtaco

### CODE REVIEW

**Comment by oconnor663** [comment]
The first commit pulls in the sigchain changes that I've already landed in libkeybase.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 0.8.23 (2015-10-20)
+
+bugfix:
+  - Import sigchain reset detection fixes from libkeybase.
```
Comment (maxtaco) [inline CHANGELOG.md]: update with revoke update

File: src/command/revoke.iced [hunk 2]
```diff
@@ -88,9 +86,13 @@ exports.Command = class Command extends Base
   run : (cb) ->
     esc = make_esc cb, "run"
     await session.login esc defer()
+    await session.get_email_or_username_i esc defer email_or_username
```
Comment (maxtaco) [inline src/command/revoke.iced]: does this actually prompt for username?

**Comment by maxtaco** [comment]
:shipit: if tests pass (and the tests hit this new endpoint)

File: src/command/revoke.iced (hunk 2 above)
Comment (oconnor663) [inline src/command/revoke.iced]: No, it gets it from the session. I think the call to `login` above guarantees that it'll be there.

**Comment by oconnor663** [comment]
The revoke test was indeed failing to test the nuke endpoint, because I wasn't using `keybase_expect` in the test code to supply the user's password properly. Fixed now.

**Pull request merged**